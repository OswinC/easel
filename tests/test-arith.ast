function int multiple(int a, int b)
{
  return a * b;
}

function int divide(int a, int b)
{
  return a / b;
}

function int mod(int a, int b)
{
  return a % b;
}

function int exp(int a, int b)
{
  return a ^ b;
}


int a;
int b;
int c;
int d;
a = multiple(2, 3);
b = divide(9, 3);
c = mod(10, 3);
d = exp(3, 4);
print(a);
print(b);
print(c);
print(d);

