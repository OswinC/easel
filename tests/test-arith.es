function int multiple(int a, int b)
{
  return a * b;
}

function int divide(int a, int b)
{
  return a / b;
}

function int mod(int a, int b)
{
  return a % b;
}

function float exp(float a, int b)
{
  return a ^ b;
}

function int main()
{
  int a;
  int b;
  int c;
  float d;
  a = multiple(2, 3);
  b = divide(9, 3);
  c = mod(10, 3);
  d = exp(3., 4);
  print(a);
  print(b);
  print(c);
  printfl(d);
  return 0;
}
