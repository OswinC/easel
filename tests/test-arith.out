6
3
1
81.000000
